.github/workflows/ci.yml

```
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    name: rake test
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version: ['3.1', '3.0', '2.7']
+        libexif-version: ['system', 'latest']
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install libexif from system
+        run: sudo apt-get install libexif-dev
+        if: ${{ matrix.libexif-version == 'system' }}
+      - name: Uninstall libexif from system
+        run: sudo apt-get remove libexif12 libexif-dev
+        if: ${{ matrix.libexif-version == 'latest' }}
+      - uses: robinraju/release-downloader@v1.3
+        name: Download latest libexif
+        if: ${{ matrix.libexif-version == 'latest' }}
+        with:
+          repository: "libexif/libexif"
+          latest: true
+          zipBall: false
+          tarBall: true
+      - name: Install latest libexif
+        if: ${{ matrix.libexif-version == 'latest' }}
+        run: |
+          sudo apt-get install autopoint
+          mkdir libexif
+          tar xvf libexif-libexif*.tar.gz -C libexif --strip-components 1
+          cd libexif
+          autoreconf -i
+          ./configure --prefix=/usr
+          make
+          sudo make install
+      - name: Set up Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+      - name: Install dependencies
+        run: bundle install
+      - name: Run tests
+        run: bundle exec rake
```

```suggestion
      - name: Run tests
        env:
          MT_CPU: $(nproc)
        run: bundle exec rake
```

We can run tests parallelly, see https://github.com/seattlerb/minitest/blob/fe3992e/lib/minitest.rb#L27